scality_sproxyd_client/sproxyd_client.py

```diff
@@ -207,23 +207,29 @@ def unexpected_http_status(response):
 
         return result
 
-    def get_meta(self, name):
-        """Open a connection and get usermd."""
-
-        def handle_200(response):
-            header = response.getheader('x-scal-usermd')
-            pickled = base64.b64decode(header)
-            return pickle.loads(pickled)
-
-        def handle_404(response):
-            return None
+    def head(self, name, headers=None):
+        """Performs a HEAD request and returns the HTTP headers."""
 
         handlers = {
-            200: handle_200,
-            404: handle_404,
+            200: lambda response: response.headers,
         }
 
-        return self._do_http('get_meta', handlers, 'HEAD', name)
+        return self._do_http('head', handlers, 'HEAD', name, headers)
+
+    def get_meta(self, name):
+        """Get the user metadata for object `name`."""
+
+        try:
+            usermd = self.head(name)['x-scal-usermd']
+        except exceptions.SproxydHTTPException as exc:
+            if exc.http_status == 404:
+                return None
+            raise
+        except KeyError:
```

Comment:
Not really. If no user metadata is set, the header x-scal-usermd will be returned, but it will be an empty string. 

The previous code would raise an EOFError on unpickling, so will the new code. I agree this is subtil/fragile. Should we catch KeyError ?